app/services/requirements/file_attachment_upload_checker.rb

```diff
@@ -63,8 +70,23 @@ def issues
   private
 
     def unsupported_type?
-      type = Marcel::MimeType.for(file, name: file.original_filename)
-      ALLOWED_FORMATS.exclude? type
+      (ALLOWED_FORMATS + [ALLOWED_ZIP]).exclude? content_type
+    end
+
+    def invalid_zip?
+      return if content_type != ALLOWED_ZIP
+
+      Zip::File.open(file) do |zf|
+        zf.each do |f|
+          type = Marcel::MimeType.for(f.get_input_stream, name: f.name)
+          return true if ALLOWED_FORMATS.exclude? type
+        end
+      end
+      false
```

Comment:
Should we have any exception handling in this method if it fails to open the zip? does Whitehall have exception handling for that? 

I figure it's possibly to have an invalid or password protected zip that can't be opened.

Reply:
From what I can see whitehall itself doesn't catch any exceptions regarding the opening of a zip. I created a new test to see if the method would bomb out with a password protected zip, but it worked correctly, as the zip can be opened to see the metadata of the files contained without actually having to open each individual (protected) file.

The other checks whitehall does is for UTF8 encoding of the filename and the contents. That the files inside the zip are one of the allowed formats as well as validating it as a ArcGIS zip but we have decided not to include this sort of check as there are very few instances of this sort of file.